### PR TITLE
docs: Magma extensions for inbound roaming

### PR DESCRIPTION
**Pull request by Arsenii-Oganov**
Signed-off-by: Arsenii Oganov <arsenii@freedomfi.com>

## Summary

Magma extensions for inbound roaming

Issue: https://github.com/magma/magma/issues/11685

### CODE REVIEW

**Comment by Arsenii-Oganov** [comment]
Issue created - https://github.com/magma/magma/issues/11685

File: docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+#### 2.1.4 Session Termination due to quota exhausted call flow
+
+The session termination procedure should be the same as is for local subscribers.
+
+![termination flow](https://user-images.githubusercontent.com/93994458/154483837-da479f5a-02e6-4147-b8ae-7647917e01a1.png)
+
+## 3 Roadmap and schedule
+
+![roadmap](https://user-images.githubusercontent.com/93994458/154483942-5a96e6e3-8aaa-4465-81a7-0eb860d8019a.png)
```
Comment (just-now) [inline docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md:63]: s/cnages/changes
better to say: s/reflects/brigns

File: docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+### 2.1 Call Flows
+
+The colour scheme in the diagram is as follows:
+Black: The existing messages for inbound roaming
+Red: The new expected messages
+Blue: The existing messages for local users which have to be added to the inbound roaming procedure
+
+#### 2.1.1 UE Attach call flow (successful case)
+
+![attach_s](https://user-images.githubusercontent.com/93994458/155352044-01e881e9-7529-4d31-8d63-0ee30b3b7291.png)
```
Comment (just-now) [inline docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md]: are svg-files available for these diagrams?
`plantuml -tsvg` should work.

To insert the image:
```![Alt text](./controllers_brief.svg)
<img src="./controllers_brief.svg">```

File: docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Magma extensions for inbound roaming
+
+Author(s): [@arsenii-oganov]
+
+Last updated: 02/18/2022
+
```
Comment (ardzoht) [inline docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md:8]: We should add the gdoc as reference here cc @uri200

**Review by ardzoht — APPROVED** [review]
lgtm, left minor comment

File: docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+Sessiond: The main change we are planning to make here is to involve sessiond in the inbound roaming call flow.
+The new logic should be added to the MME task:
+Before sending Create Session Create request to S8 Proxy the MME should send “Create Session” message to the Sessiond. This message will be trigger to generate CCR-I message towards session-proxy with first volume quota request. When the quota has been received from OCS the sessiond should send "Create Session Response" towards MME to allow S8 crate session procedure. In case of no quota has been received from OCS the sessiond should trigger session termination to the MME.
+Once MME received Create Bearer Request and Table0 has been provisioned to the OVS the MME should sent "Final Create Session Request" towards sessiond to start get stats procedure from OVS. After reaching the configurable volume threshold on sessiond the sessiond should generate CCR-U to ask the next quota in OCS. This is an existing procedure for home subscribers but we should add the procedure to the inbound roaming call flow.
+
+### 2.1 Call Flows
+
+The colour scheme in the diagram is as follows:
```
[the remark this answers is not in the document]
Comment (uri200) [reply docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md:34]: lets fix those issues or it may not be able to merge

**Review by electronjoe — APPROVED** [review]
LGTM @ Alex's asks

**Comment by ekowtaylor** [comment]
@Arsenii-Oganov do you still want to move forward with this change? Then please rebase and fix any other failing checks

**Comment by ardzoht** [comment]
We can merge this PR with Oriol's approval after pending changes are reflected here cc @uri200

File: docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ardzoht) [reply docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md:34]: cc @Arsenii-Oganov

File: docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md (hunk 3 above)
Comment (uri200) [reply docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md:8]: Can you add the google doc with the design here please

File: docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+Red: The new expected messages
+Blue: The existing messages for local users which have to be added to the inbound roaming procedure
+
+#### 2.1.1 UE Attach call flow (successful case)
+
+![attach](https://user-images.githubusercontent.com/93994458/154483466-712c211d-7754-47b2-9993-4dee2da5d052.png)
+
+#### 2.1.2 UE Attach call flow (unsuccessful case)
+
+The behaviour if quota has not been received from OCS.
```
[the remark this answers is not in the document]
Comment (uri200) [reply docs/readmes/proposals/p026_magma_inbound_roaming_extentions.md:45]: fix this too

**Pull request merged**